### DISCUSSION TITLE
How does this extension communicates with the underlying compiler?

### DISCUSSION THREAD

**Discussion by famastefano** [microsoft/vscode-cpptools#10411] [Q&A] [answered]
Because I wanted to add features like Unreal Engine specific light bulb suggestions, intellisense for macros like UCLASS/UPROPERTY etc.
I can fork the clangd VSCode extension, but it's very barebones, and this one is the de facto standard for C++ development.
I don't understand if I can "simply" modify the underlying compiler (clang) and/or language server (clangd) and everything will suddenly JustWork™️.
This extension uses a clangd-like approach, where they start the language server and then communicates with it as a Client, or you've implented your own language server that somehow interfaces with the compiler?

**Comment by dev-joss** [comment, marked as answer]
Uses a program called cpptools.exe which is closed source as far as I understand

**Reply by sean-mcmanus** [reply]
@eclazi That is correct.

**Reply by famastefano** [reply]
Thank you for the answer.
I guess the only way to "improve" it on the backend side will be to join MS ;)

**Reply by sean-mcmanus** [reply]
@famastefano You could also try filing a bug or feature request on VS (and we may inherit the fix from them) -- I found a similar bug report at https://developercommunity.visualstudio.com/t/Intellisense-for-Macro-Specifiers-and-li/738774 but it looks like the reporter didn't provide enough info for the triagers to understand.
